### ISSUE TITLE
phpunit 404: Not found with v2.25.3

### ISSUE THREAD

**Issue by blizzz** [shivammathur/setup-php#743]
**Describe the bug**

With today's v2.25.3 the installation of phpunit:9 fails. Going back to 2.25.2 has it succeeding.

**Version**
- [X] I have checked releases, and the bug exists in the latest patch version of `v1` or `v2`.
- [X] `v2`
- [ ] `v1`

**Runners**

- [X] GitHub Hosted
- [ ] Self Hosted

**Operating systems**
Ubuntu-20.04

**PHP versions**
8.0, 8.1, 8.2

**To Reproduce**

```
name: PHPUnit foobar

on: pull_request

permissions:
  contents: read

concurrency:
  group: phpunit-foobar-${{ github.head_ref || github.run_id }}
  cancel-in-progress: true

jobs:
  phpunit-foobar:
    runs-on: ubuntu-20.04

    strategy:
      matrix:
        php-versions: ['8.0', '8.1', '8.2']

    steps:
      - name: Checkout server
        uses: actions/checkout@v3
        with:
          submodules: true

      - name: Set up php ${{ matrix.php-versions }}
        uses: shivammathur/setup-php@v2
        with:
          php-version: ${{ matrix.php-versions }}
          extensions: ctype, curl, dom, fileinfo, gd, imagick, intl, json, mbstring, oci8, openssl, pcntl, pdo_sqlite, posix, sqlite, xml, zip
          tools: phpunit:9
          coverage: none
          ini-file: development
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

```

**Expected behavior**

In the Set Up PHP 8.x step we  expect phpunit to be added correctly, like:

```
✓ phpunit Added phpunit 9.6.9
```

**Screenshots/Logs**

![Screenshot_20230612_201624](https://github.com/shivammathur/setup-php/assets/2184312/284b1b10-c535-47d5-8ccd-fa6f82b6f504)

**Additional context**

When we use setup-php@c5fc0d8281aba02c7fda07d3a70cc5371548067d (2.25.2) we get the expected behaviour.

**Are you willing to submit a PR?**
Not familiar with typescript or writing actions :(

**Comment by gman-wa** [comment]
change 

tools: phpunit:9

to

tools: phpunit

it will pull the correct version of phpunit for the version of php you are running per the commit last night

https://github.com/shivammathur/setup-php/commit/4630172078882a40b6f1bc2ce433c662a43339be

the readme here needs to be updated.

**Comment by blizzz** [comment]
> change
> 
> tools: phpunit:9
> 
> to
> 
> tools: phpunit
> 
> it will pull the correct version of phpunit for the version of php you are running per the commit last night
> 
> [4630172](https://github.com/shivammathur/setup-php/commit/4630172078882a40b6f1bc2ce433c662a43339be)
> 
> the readme here needs to be updated.

I am doubtful this is the intention. The version, if specified, is still taken into consideration. And fetching https://phar.phpunit.de/phpunit-9.phar, for instance, would always redirect to the latest phpunit v9 release phar.

**Comment by SystemKeeper** [comment]
Requested URL: https://api.github.com/repos/phpunit/phpunit/git/matching-refs/tags%2F9. -> results in 404
Probably should be: https://api.github.com/repos/sebastianbergmann/phpunit/git/matching-refs/tags%2F9. ? At least that works.

**Comment by blizzz** [comment]
> Requested URL: https://api.github.com/repos/phpunit/phpunit/git/matching-refs/tags%2F9. -> results in 404 Probably should be: https://api.github.com/repos/sebastianbergmann/phpunit/git/matching-refs/tags%2F9. ? At least that works.

Aye, the repository was changed in that commit, this fits.

**Comment by shivammathur** [comment]
@blizzz @SystemKeeper @gman-wa 
Sorry for the regression.

Fixed in [`2.25.4`](https://github.com/shivammathur/setup-php/releases/tag/2.25.4)

**Comment by blizzz** [comment]
No worries, thank you for the quick fix @shivammathur :)
